.travis.yml

```diff
@@ -52,18 +52,33 @@ matrix:
       env:
         - DEPS=latest
 
+before_install:
+  - shopt -s expand_aliases
+  - alias composer="travis_retry docker-compose run --rm php composer"
+  - alias php="travis_retry docker-compose run -e TRAVIS -e TRAVIS_JOB_ID --rm php php"
+  - docker-compose build --build-arg PHP_VERSION=${TRAVIS_PHP_VERSION} --build-arg XDEBUG=${TEST_COVERAGE:+1} --no-cache php
+  - composer --version
+  - php -v
+  - php -m
+
 install:
+  - php --info
   - composer install $COMPOSER_ARGS --ignore-platform-reqs
+  - if [[ $LEGACY_DEPS != '' ]]; then composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer config "platform.ext-mongo" "1.6.16" && composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
-
-  - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer require --dev $COMPOSER_ARGS $ADAPTER_DEPS
+  - if [[ $TEST_COVERAGE == 'true' ]]; then composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - travis_retry composer config "platform.ext-mongo" "1.6.16" && composer require --dev $COMPOSER_ARGS 
```

Comment:
The first part of this seems unnecessary because of https://github.com/doctrine/DoctrineMongoODMModule/blob/24e83aeeafc39e20f3b80b101cfc05d8a3ca8a0e/composer.json#L99-L101

Reply:
You're right.  I have to remove travis_retry from all composer calls because they are aliased and it works.

Reply:
https://github.com/doctrine/DoctrineMongoODMModule/pull/214/files#diff-354f30a63fb0907d4ad57269548329e3L57